### ISSUE TITLE
[beta4] Found ugly output when trying to debug an invalid opcode in a constructor

### ISSUE THREAD

**Issue by roderik** [trufflesuite/truffle#576]
I added a new require in a constructor, and it failed. I was trying to use the new debugger to figure out which one (and to just test it out :)) and this was the result:

```
$ truffle develop
Running development blockchain at http://localhost:9545/...
Truffle Develop started.

truffle(develop)> migrate
Using network 'develop'.

Running migration: 1_initial_migration.js
  Deploying Migrations...
  ... 0x1f33f14077b7837051d1b8002ad6ec7c654a722dbab0dac94a1e6708818cd9ac
  Migrations: 0x3ed10fd31b3fbb2c262e6ab074dd3c684b8aa06b
Saving successful migration to network...
  ... 0x429a40ee574664a48753a33ea0c103fc78c5ca7750961d567d518ff7a31eefda
Saving artifacts...
Running migration: 2_deploy_contracts.js
  Running step...
  Deploying MiniMeTokenFactory...
  ... 0xce6b06adbecb66cae231ccc6285022f94dd7f4a60a4ae7da076f9993c9dc4851
  MiniMeTokenFactory: 0x377bbcae5327695b32a1784e0e13bedc8e078c9c
  Deploying MultiSigWallet...
  ... 0x7ed0834e848532c113f9ec773f56f0d7cc061bbac8c27fea805bce1be97a0030
  MultiSigWallet: 0xf070d8253dd7efe4b20e5299b6a000b0b9cbb261
  Deploying DataBrokerDaoToken...
  ... 0xa78b226761c0f9637f9881629e2705046ddb4855562996cf1d625cab5bf3a317
  DataBrokerDaoToken: 0x62b5df64d30d8100d3578130ee6682caba9e2107
  Deploying EarlyTokenSale...
  ... 0x44c7f12e42ed41883f24f3d3c5da797a7380fbca3238fa7157a01c453dd2736e
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Error: VM Exception while processing transaction: invalid opcode
    at Object.InvalidResponse (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:44006:16)
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:359607:36
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:86262:11
    at /usr/local/lib/node_modules/truffle/build/cli.bundled.js:322668:9
    at XMLHttpRequest.request.onreadystatechange (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:358306:7)
    at XMLHttpRequestEventTarget.dispatchEvent (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:71245:18)
    at XMLHttpRequest._setReadyState (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:71535:12)
    at XMLHttpRequest._onHttpResponseEnd (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:71690:12)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:71650:24)
    at emitNone (events.js:110:20)
    at IncomingMessage.emit (events.js:207:7)
    at endReadableNT (_stream_readable.js:1059:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickDomainCallback (internal/process/next_tick.js:218:9)
truffle(develop)> debug 0x44c7f12e42ed41883f24f3d3c5da797a7380fbca3238fa7157a01c453dd2736e
Gathering transaction data...

Addresses affected:
  0x25551a8a7404dd909330a282b2403a239f8aab4c - ?

Warning: The source code for one or more contracts could not be found.

Commands:
(enter) last command entered (step next)
(o) step over, (i) step into, (u) step out, (n) step next
(;) step instruction, (p) print instruction, (h) print this help, (q) quit


? | 0x25551a8a7404dd909330a282b2403a239f8aab4c:

1: // No source code found.

debug(develop:0x44c7f12e...)> o
TypeError: Cannot read property 'start' of undefined
    at Debugger.step (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333165:58)
    at Debugger.stepOver (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333295:31)
    at Object.interpreter (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:327749:22)
    at ReplManager.interpret (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:200019:18)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:115:13)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:279:10)
    at REPLServer.Interface._line (readline.js:626:8)
    at REPLServer.Interface._ttyWrite (readline.js:906:14)
    at REPLServer.self._ttyWrite (repl.js:509:7)
    at ReadStream.onkeypress (readline.js:158:10)
    at emitTwo (events.js:125:13)
    at ReadStream.emit (events.js:213:7)
debug(develop:0x44c7f12e...)> ;
TypeError: Cannot read property 'name' of undefined
    at Debugger.isCall (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333388:21)
    at Debugger.advance (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333091:12)
    at Object.interpreter (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:327761:22)
    at ReplManager.interpret (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:200019:18)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:115:13)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:279:10)
    at REPLServer.Interface._line (readline.js:626:8)
    at REPLServer.Interface._ttyWrite (readline.js:906:14)
    at REPLServer.self._ttyWrite (repl.js:509:7)
    at ReadStream.onkeypress (readline.js:158:10)
    at emitTwo (events.js:125:13)
    at ReadStream.emit (events.js:213:7)
debug(develop:0x44c7f12e...)> 
TypeError: Cannot read property 'start' of undefined
    at Debugger.step (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333165:58)
    at Object.interpreter (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:327758:22)
    at ReplManager.interpret (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:200019:18)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:115:13)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:279:10)
    at REPLServer.Interface._line (readline.js:626:8)
    at REPLServer.Interface._ttyWrite (readline.js:906:14)
    at REPLServer.self._ttyWrite (repl.js:509:7)
    at ReadStream.onkeypress (readline.js:158:10)
    at emitTwo (events.js:125:13)
    at ReadStream.emit (events.js:213:7)
    at emitKeys (internal/readline.js:420:14)
debug(develop:0x44c7f12e...)> 
TypeError: Cannot read property 'start' of undefined
    at Debugger.step (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:333165:58)
    at Object.interpreter (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:327758:22)
    at ReplManager.interpret (/usr/local/lib/node_modules/truffle/build/cli.bundled.js:200019:18)
    at bound (domain.js:301:14)
    at REPLServer.runBound [as eval] (domain.js:314:12)
    at REPLServer.onLine (repl.js:440:10)
    at emitOne (events.js:115:13)
    at REPLServer.emit (events.js:210:7)
    at REPLServer.Interface._onLine (readline.js:279:10)
    at REPLServer.Interface._line (readline.js:626:8)
    at REPLServer.Interface._ttyWrite (readline.js:906:14)
    at REPLServer.self._ttyWrite (repl.js:509:7)
    at ReadStream.onkeypress (readline.js:158:10)
    at emitTwo (events.js:125:13)
    at ReadStream.emit (events.js:213:7)
    at emitKeys (internal/readline.js:420:14)
```

Things that seem off on first glance:

* why couldn't it find the source? I executed all the upgrade steps like described.
* and if it cannot find the source, there were lots of javascript errors that should not be there

## Environment

* Operating System: OSX
* Truffle version: 4.0.0.beta0
* Ethereum client: built in testrpc
* node version: 8.5
* yarn: 1.0.2

**Comment by tcoulter** [comment]
Thanks for this report. We'll attempt to reproduce, and may come back to you with more questions if we have trouble. Is your code open-source (i.e., freely available online)?

**Comment by roderik** [comment]
Here you go: https://github.com/DataBrokerDAO/crowdsale-contracts

**Comment by gnidan** [comment]
@roderik I am having trouble reproducing this issue. Has it gone away, or do you have a specific branch that I try?

I do notice that your migration `3_different_tokensale.js` is failing within `truffle develop` due to a hardcoded address. That use case doesn't work with Truffle Develop, but I don't think that's the error you're describing above.

Thanks!

**Comment by roderik** [comment]
Hi @gnidan, the error was fixed already so I have no test branch. The issue is that for some reason it could not find source code. 

Ok to close for now, will report (with test repos) any other errors i find :)

**Comment by gnidan** [comment]
Thanks @roderik!

It is possible that the problem was due to breaking changes in the Truffle 4 contract JSON files. We've noticed that there have been some problems with compatibility. This error may have disappeared in the course of those artifacts being recreated or overwritten.

**Comment by thesved** [comment]
I could reproduce this issue: you should deploy a contract with an argument which violates a requirement in the constructor.

`contract Test {
    function Test(bool _val) {
        require(_val);
    }
}`

When you deploy, like
`deployer.deploy(Test, false);`

This obviously will violate the requirement, and when you try to debug the transaction it won't find the source code.

**Comment by ppoliani** [comment]
I've encountered the same issue. In my case, I follow a different approach. To be more specific:

- Deploy some libraries using truffle migrations 
- Use web3.js to deploy another contract

When deploying the contract using web3 I can see the contract in the ganache-cli logs

<img width="675" alt="screen shot 2018-02-06 at 18 46 03" src="https://user-images.githubusercontent.com/1389619/35878040-fd43bfe6-0b6e-11e8-9078-4cd243c599a0.png">

I try then to send a transaction via truffle console to call a smart contract method:

<img width="1050" alt="screen shot 2018-02-06 at 18 48 11" src="https://user-images.githubusercontent.com/1389619/35878081-1d294e98-0b6f-11e8-9434-4bfe36a80ccb.png">

I get an error:

<img width="674" alt="screen shot 2018-02-06 at 18 48 24" src="https://user-images.githubusercontent.com/1389619/35878126-3dc3ffd6-0b6f-11e8-9481-ee80ce0365fa.png">

But that's fine, it can be due to a bug in the smart contract. However, when I try to debug the using the above txHash I get the following error

<img width="719" alt="screen shot 2018-02-06 at 18 48 57" src="https://user-images.githubusercontent.com/1389619/35878171-58763a1a-0b6f-11e8-84b8-c608e9752ff3.png">

It looks like it cannot find the source code of the contract I manually deployed via JavaScript and web3.js. 

The thing is that the ganache logs (look screenshot above) showed that the contract was deployed. Also, when I investigate the contract instance in the truffle console

`Contract.at('0xE6099d65d9a40A9dee1908f82C9Ecb303056077c')`

I can see in the bytecode that the libraries I linked via solc are not actually there. That is, I can see the placeholders but no the addresses.

e.g. `__FeesLib_______________________________`

I know there are a few points here :) But it would be nice to know if we can manually deploy contracts to ganache without using `truffle migrate`
